src/app/routes/frontend.server.routes.ts

```diff
@@ -0,0 +1,52 @@
+import { celebrate, Joi, Segments } from 'celebrate'
+import { Router } from 'express'
+
+import * as FrontendServerController from '../controllers/frontend.server.controller'
+import { GoogleAnalyticsFactory } from '../factories/google-analytics.factory'
+
+export const FrontendRouter = Router()
+
+/**
+ * Generate the templated Javascript code for the frontend to initialise Google Tag Manager
+ * Code depends on whether googleAnalyticsFeature.isEnabled
+ * @route GET /frontend/datalayer
+ * @return 200 when code generation is successful
+ * @return 400 when code generation fails
+ */
+FrontendRouter.get('/frontend/datalayer', GoogleAnalyticsFactory.datalayer)
```

Comment:
all these routes can have their `/frontend` prefix removed, and you can add the `/frontend` prefix in `src/loaders/express/index.ts`. this is so that we can establish a clear hierarchy of APIs

Reply:
shifted